mllib/src/main/scala/org/apache/spark/ml/classification/Classifier.scala

```diff
@@ -103,7 +103,10 @@ abstract class Classifier[
    * @throws IllegalArgumentException  if metadata does not specify numClasses, and the
    *                                   actual numClasses exceeds maxNumClasses
    */
-  protected def getNumClasses(dataset: Dataset[_], maxNumClasses: Int = 100): Int = {
+  protected def getNumClasses(dataset: Dataset[_],
```

Comment:
Since we don't have Instrumentation readily available, I recommend using the old logging here.  (That will also avoid this API-breaking change which is causing the MiMA failure.)